crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -4650,6 +4650,15 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
             targets,
         } = delete;
         for target in targets {
+            // Note that `del x` marks `x` IS_BOUND in the current scope in the
+            // SemanticIndexBuilder (unless `x` is already marked nonlocal or global), so invalid
+            // examples like this give an error:
+            //
+            // ```py
+            // x = 1
+            // def foo():
+            //      del x  # error: [unresolved-reference]: Name `x` used when not defined
+            // ```
```

Comment:
I think this comment is probably not helpful here. I understand why it made sense in the context of figuring out how to implement this, but it doesn't directly relate to any code here, and I don't know that it will be helpful to a future reader.